crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -706,28 +689,78 @@ impl AuthorityStorePruner {
                 .unwrap_or_default() as i64,
         );
 
-        tokio::task::spawn(async move {
-            loop {
-                tokio::select! {
-                    _ = objects_prune_interval.tick(), if config.num_epochs_to_retain != u64::MAX => {
-                        if let Err(err) = Self::prune_objects_for_eligible_epochs(&perpetual_db, &checkpoint_store, rpc_index.as_deref(), pruner_db.as_ref(), config.clone(), metrics.clone(), epoch_duration_ms).await {
-                            error!("Failed to prune objects: {:?}", err);
+        #[cfg(tidehunter)]
+        {
+            if let Some(num_epochs_to_retain) = config.num_epochs_to_retain_for_checkpoints() {
+                tokio::task::spawn(async move {
+                    loop {
+                        objects_prune_interval.tick().await;
+                        if let Err(err) = Self::prune_th(
+                            &perpetual_db,
+                            &checkpoint_store,
+                            num_epochs_to_retain,
+                            pruner_watermark.clone(),
+                        ) {
+                            error!("Failed to prune tidehunter: {:?}", err);
                         }
-                    },
-                    _ = checkpoints_prune_interval.tick(), if !matches!(config.num_epochs_to_retain_for_checkpoints(), None | Some(u64::MAX) | Some(0)) => {
-                        if let Err(err) = Self::prune_checkpoints_for_eligible_epochs(&perpetual_db, &checkpoint_store, rpc_index.as_deref(), pruner_db.as_ref(), config.clone(), metrics.clone(), epoch_duration_ms).await {
-                            error!("Failed to prune checkpoints: {:?}", err);
+                    }
+                });
+            }
+        }
+        #[cfg(not(tidehunter))]
+        {
+            let mut checkpoints_prune_interval =
+                tokio::time::interval_at(Instant::now() + pruning_initial_delay, tick_duration);
+            let mut indexes_prune_interval =
```

Comment:
We still need to run index pruner when tidehunter is configured, since index db does not use tidehunter?